icgcget/commands/download.py

```diff
@@ -29,132 +30,131 @@
             self.gt_client.session = session
-            uuids = self.get_uuids(object_ids['cghub'])
+            uuids = self.get_uuids(file_data['cghub'])
             return_code = self.gt_client.download(uuids, cghub_key, cghub_path, staging, cghub_transport_parallel)
-            object_ids = self.icgc_client.session
-            self.check_code('Cghub', return_code)
-            self.move_files(staging, output)
+            session = self.cleanup('CGHub', return_code, staging, output, self.gt_client)
 
-        if 'aws-virginia' in object_ids and object_ids['aws-virginia']:
-            check_access(self, icgc_token, 'icgc', icgc_path)
+        if 'aws-virginia' in file_data and file_data['aws-virginia']:
+            check_access(self, icgc_token, 'icgc', self.icgc_client.docker, icgc_path)
             self.icgc_client.session = session
-            uuids = self.get_uuids(object_ids['aws-virginia'])
+            uuids = self.get_uuids(file_data['aws-virginia'])
             return_code = self.icgc_client.download(uuids, icgc_token, icgc_path, staging, icgc_transport_parallel,
                                                     file_from=icgc_transport_file_from, repo='aws')
-            object_ids = self.icgc_client.session
-            self.check_code('Icgc', return_code)
-            self.move_files(staging, output)
+            session = self.cleanup('ICGC', return_code, staging, output, self.icgc_client)
 
-        if 'ega' in object_ids and object_ids['ega']:
-            check_access(self, ega_username, 'ega', ega_path, ega_password, udt=ega_udt)
+        if 'ega' in file_data and file_data['ega']:
+            check_access(self, ega_username, 'ega', self.ega_client.docker, ega_path, ega_password, udt=ega_udt)
             if ega_transport_parallel != '1':
-                self.logger.warning("Parallel streams on the ega client may cause reliability issues and failed " +
+                self.logger.warning("Parallel streams on the EGA client may cause reliability issues and failed " +
                                     "downloads.  This option is not recommended.")
             self.ega_client.session = session
-            uuids = self.get_uuids(object_ids['ega'])
+            uuids = self.get_uuids(file_data['ega'])
             return_code = self.ega_client.download(uuids, ega_username, ega_path, staging, ega_transport_parallel,
                                                    ega_udt, password=ega_password)
-            object_ids = self.icgc_client.session
-            self.check_code('Ega', return_code)
-            self.move_files(staging, output)
+            session = self.cleanup('EGA', return_code, staging, output, self.ega_client)
 
-        if 'collaboratory' in object_ids and object_ids['collaboratory']:
-            check_access(self, icgc_token, 'icgc', icgc_path)
+        if 'collaboratory' in file_data and file_data['collaboratory']:
```

Comment:
The way it currently works is that a repo key is only added to object_ids if that repo is on the manifest.  Calling just the second expression led to an invalid key error.  However, there were issues where all the ids got removed from from a given repo after the key was added because the files were found in the local directory. So it's also necessary to check that there are values under that key